nd4j/nd4j-backends/nd4j-api-parent/nd4j-api/src/main/java/org/nd4j/linalg/api/ops/impl/layers/recurrent/weights/RNNWeights.java

```diff
@@ -29,7 +32,33 @@
         return res;
     }
 
+    protected static INDArray[] filterNonNull(INDArray... args){
+        int count = 0;
+        for(INDArray v : args){
+            if(v != null){
+                count++;
+            }
+        }
+
+        INDArray[] res = new INDArray[count];
+
+        int i = 0;
+
+        for(INDArray v : args){
+            if(v != null){
+                res[i] = v;
+                i++;
+            }
+        }
+
+        return res;
+    }
```

Comment:
Rather than adding a whole new overload with code duplication, let's make the existing method generic, so that it works for both SDVariable and INDArray.
See DynamicCustomOp.wrapFilterNull for how